src/main/java/com/example/booksstore/dto/BookRequestDto.java

```diff
@@ -1,13 +1,23 @@
 package com.example.booksstore.dto;
 
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.PositiveOrZero;
 import java.math.BigDecimal;
 import lombok.Data;
+import org.hibernate.validator.constraints.ISBN;
 
 @Data
 public class BookRequestDto {
+    @NotBlank
     private String title;
+    @NotBlank
     private String author;
+    @NotBlank
+    @ISBN
     private String isbn;
```

Comment:
I think `isbn` field is not marked with `notNull` on the entity side, please double-check

Reply:
also it is actually marked as `unique` and we do not want to return a raw SQL constraint violation exception, so what we might want to do is add a ISBN uniqueness check to the service on create/update

Reply:
![image](https://github.com/Recdt/Books-Store/assets/90136583/a7c52ce3-9995-4603-9f4d-6ddb42ec51aa)
Is this not a not null check or you are talking about @NotNull annotation? But do we need 2 same checks? 
And do we need this check on service layer if in situation when we are trying to add element with the same isbn we are making this check in DB?

Reply:
It's better to ask the questions on the platform because it is easy to lose them here :)
 * Column in the entity is NotNull, so yes we can have `@NotBlank` here, I was just thinking maybe it was `nullable = true` but it is not.
 * Yes we need to add a check if the ISBN that we have received in the request DTO is not present in the DB already(or equal to the one that this book already has in case of an update) and throw a custom exception if it already exists